main/lsp/LSPTypecheckerCoordinator.h

```diff
@@ -48,21 +51,34 @@ class LSPTypecheckerCoordinator final {
      * Initializes typechecker and runs typechecking for the first time.
      * TODO(jvilk): Make non-blocking when we implement preemption.
      */
-    void initialize(LSPFileUpdates initializeUpdate);
+    void initialize(std::unique_ptr<InitializedTask> initializedTask);
 
     /**
-     * Typechecks the given updates on the slow path. Blocks until the typechecking actually begins.
+     * Runs the given typecheck task asynchronously on the slow path.
      */
-    void typecheckOnSlowPath(LSPFileUpdates updates);
+    void typecheckOnSlowPath(std::unique_ptr<SorbetWorkspaceEditTask> typecheckTask);
 
     /**
-     * Schedules a task on the typechecker thread, and blocks until `task` completes. If the task has
-     * `enableMultithreaded` set to "true", then the given task is allowed to use the full threadpool at the cost of not
-     * being able to preempt slow path typechecking.
-     * TODO(jvilk): Make single-threaded tasks scheduled this way preempt the slow path.
+     * Schedules a task to run on the typechecker thread. Blocks until it completes.
      */
     void syncRun(std::unique_ptr<LSPTask> task);
```

Comment:
syncRun used to decide if a task should preempt. That design no longer makes sense, since we need to determine if preemption is possible _before_ scheduling a specific task to run.

LSPLoop now determines whether or not to schedule tasks to preempt, and calls the new `trySchedulePreemption` method.